### ISSUE TITLE
[feature]: Api Key based authentication

### ISSUE THREAD

**Issue by mcroissant** [hoppscotch/hoppscotch#2000]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Hello,

We see many apis and api gateways using api key as a standard authentication mechanism, it would be awesome to support this authentication method in hoppscotch. This would require the ability to specificy the place where to set the api key (header or path) and under which name (different vendor and implementation use different header name).

### Why should this be worked on?

Currently APIs using API Keys passed through headers have their authentication specified as any other header when they actually concern Authentication. To have a consistent handling of Authentication it should be made possible to specify API Key as an autentication method in hoppscotch

**Comment by EdwinFajardoBarrera** [comment]
I would like to work in this issue, can you assign it to me?

**Comment by EdwinFajardoBarrera** [comment]
I have some questions about the implementation:

I've implemented the capability of adding API_KEY-based authentication, adding it in the specified header or by query params.

In the first case, I used the `effectiveFinalHeaders` function in `EffectiveURL.ts`

In the second case, I'm pushing a new element to params. I tried many options but those have many issues:

- Adding new param by request.param attribute
`request.params.pop()
  request.params.push({
    active: true,
    key,
    value: parseTemplateString(value, envVariables),
  })`
In this way, I need to remove the last param (empty one) and push the new one. This is the way it works better but it haves inconsistency in the Params tab: Activate/Deactivate and delete buttons don't work properly.

- Adding new param by addRESTParam method in store
It doesn't set the param before the request is sent.

I really would like to read your comments of architectonical guidelines. Thanks in advance
